### PR TITLE
Fix redundant merge status updates

### LINKED ISSUE #17948: Redundant merge status updates
### The problem

In the merge dialog,When I first select a branch with commit (or a branch with conflicting files), and then select a branch with clean status again ，the merge status will flash once,the flash content is clean merge message with commit count

### Release version

Version 3.3.6 (arm64)

### Operating system

Macos14.0

### Steps to reproduce the behavior

_No response_

### Log files

_No response_

### Screenshots

You can reproduce by slowing down the recording speed
Kapture 2024-01-02 at 11 46 36

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GengShengJia**
Closes #17948 

## Description

This PR fixes a flicker of merge information when switching branches in the merge branch dialog box to complete merge checks.

This is a completed case: when you first select a branch with a merge commit or a branch with conflicting files, and then switch to a clean state branch again, the merge preview will flash once.

image

There are three calls to updateMergeStatusPreview in this code snippet. Deleting the second call solves this problem. Deleting the second function call seems to have no effect, ` currentBranch= Null ` will also call the update function once at the end of the function.
### Screenshots

image

## Release notes

Notes: [Fixed] Merge branch dialog no longer shows flickering merge preview when switching branches

### CODE REVIEW

**Review by tidy-dev — CHANGES_REQUESTED** [review]
@GengShengJia I believe this section of code a little hard to follow.

The mergeStatus can be:
1. null
2. loading
3. clean
4. invalid
5. conflicted

When update Status is called:
1. Sets the merge status to loading and runs updateMerges status to show loading indication.
2. Attempts to determine mergability. 
- This can result in clean, invalid, or conflicted. Clean is the only one that uses the commit count. Thus, if it is conflicted or invalid, going ahead with the update of UI here makes sense. 

I would propose to keep the update statment here but, add a check for if merge status is invalid or conflicted, then do the  update and return (skip the commit count and other update statement).  Also add in a comment stating that the clean status is the only one that needs the ahead/Behind count.

**Comment by GengShengJia** [comment]
Thanks for your answer, deleting the code here is indeed an unwise choice, I know what to do now

**Review by tidy-dev — APPROVED** [review]
Looks great! Thank you for the improvement!
